### ISSUE TITLE
Bottom toolbar expanding above the keyboard

### ISSUE THREAD

**Issue by radim10** [tneotia/html-editor-enhanced#32]
**Problem**
When the user selects some text and wants to make it for example Heading 2 style, the dropdown menu expands behind the opened keyboard. This is not much usable, because the user can not select the options which are behind the keyboard.

**Solution**
It would be better if the whole dropdown menu would expand above the keyboard and if there is not enough space it would be a scrollable dropdown list or something like that.

![image](https://user-images.githubusercontent.com/62602887/116815653-cd071100-ab5e-11eb-86e8-dacb4419007b.png)

**Comment by tneotia** [comment]
Thanks for this report, I'm working on a fix. I'm not super experienced with Flutter layouts so any help would be appreciated! 

It seems the scaffold doesn't resize itself when the keyboard is up so the dropdown thinks it has the whole screen to display the menu when actually half the screen is taken by the keyboard.

**Comment by tneotia** [comment]
Quick update: With a custom implementation and some work with rectangles and sizing I can get this result:

<img src="https://user-images.githubusercontent.com/50850142/117051154-382e2000-ace4-11eb-8206-c1fca7e620bd.png" width="250px"/>

Not the best but it is progress.

**Comment by radim10** [comment]
> Quick update: With a custom implementation and some work with rectangles and sizing I can get this result:
> 
> <img src="https://user-images.githubusercontent.com/50850142/117051154-382e2000-ace4-11eb-8206-c1fca7e620bd.png" width="250px"/>
> 
> Not the best but it is progress.

This looks promising.

**Comment by tneotia** [comment]
It looks good now when keyboard is down but when keyboard is up the menu isn't shifted up...

| Down | Up |
| - | - |
|  ![image](https://user-images.githubusercontent.com/50850142/117175102-7555e900-ad9c-11eb-9925-f026f3fd0867.png) |  ![image](https://user-images.githubusercontent.com/50850142/117175031-62dbaf80-ad9c-11eb-90b5-b33fa79a428c.png) |

**Comment by radim10** [comment]
I think that it would be better to make the menu scrollable and in smaller height for usability purposes or make it horizontal like the toolbar with h1 h2 ... next to each other.

**Comment by tneotia** [comment]
> it would be better to make the menu scrollable and in smaller height for usability purposes

Yes I agree, this is the end goal, first I have to make sure the menu shifts properly when the keyboard is up otherwise the original problem will not be solved ;) 

> make it horizontal like the toolbar with h1 h2 ... next to each other, like two-step toolbar.

I like this idea also but then I have to make sure the webview editor area shifts up so the two-step is visible, and this transition won't be very smooth because it is not synced with Flutter (we have to use JavaScript to reduce editor height). That's why I am trying to focus on the first method you suggested so the UX is good.

Apologies it is taking a long time, I haven't had much time to work on this. Hopefully by the weekend I will have a good solution for this problem. In the meantime you can use `ToolbarPosition.aboveEditor` as a workaround.

**Comment by tneotia** [comment]
I think I've gotten it to a satisfactory state. Let me know your thoughts - if you have any further ideas I can look into them before pushing the code to `master`. (If not then still please respond so I can get the "OK" to push this code and release 2.1.0) :)


https://user-images.githubusercontent.com/50850142/117551134-274a1b00-b012-11eb-82b5-d7edfb718cf0.mp4

**Comment by radim10** [comment]
> I think I've gotten it to a satisfactory state. Let me know your thoughts - if you have any further ideas I can look into them before pushing the code to `master`. (If not then still please respond so I can get the "OK" to push this code and release 2.1.0) :)
> 
>  Reversed.Dropdown.mp4

This is the solution I wanted. You can push to master and please make sure that this solution is applied to all dropdown menus in the toolbar. Thanks for your work, I appreciate it.

**Comment by tneotia** [comment]
Pushed to v2.1.0. Thanks!

**Comment by radim10** [comment]
Getting this error after updating to v2.1.0.
/C:/flutter/.pub-cache/hosted/pub.dartlang.org/html_editor_enhanced-2.1.0/lib/utils/utils.dart:288:57: Error: The method 'copyWith' isn't defined for the class 'ScrollBehavior'.
 - 'ScrollBehavior' is from 'package:flutter/src/widgets/scroll_configuration.dart' ('/C:/flutter/packages/flutter/lib/src/widgets/scroll_configuration.dart').
Try correcting the name to the name of an existing method, or defining a method named 'copyWith'.
              behavior: ScrollConfiguration.of(context).copyWith(

**Comment by aniketsongara** [comment]
@tneotia  Great to see the issue has been resolved. 
here I have something similar with this issue is, when inserting link of any files from the editor toolbar options 

https://user-images.githubusercontent.com/59816309/117924870-12bc9a00-b314-11eb-9816-0713f9e9f76b.mp4

**Comment by tneotia** [comment]
Please file a separate issue and I will take a look, thanks :)
